_docs/for-contributors/2017-01-01-building-from-source.md

```diff
@@ -21,14 +21,22 @@ Electron-based wallet called “Daedalus”.
 The source code for both Cardano SL and Daedalus Bridge can be obtained
 from the [official repository](https://github.com/input-output-hk/cardano-sl).
 
-The [Haskell Tool Stack](https://haskellstack.org) is required to build Cardano SL. Furthermore, we strongly suggest using [Nix package manager](https://nixos.org/nix/download.html) to get the correct dependencies for building Cardano SL. It will fetch the correct `openssl` version, but won't override the system-installed version. The following commands assume that you already has `stack` and `nix-*` programs.
+Cardano SL supports three ways for building itself:
+
+-   Pure [Nix](https://nixos.org/nix/)-based (backed by a build cache from the IOHK CI)
+-   Mixed Stack+Nix mode (also cache-backed)
```

Comment:
- [Stack](https://github.com/commercialhaskell/stack) with Nix for system libraries

Reply:
Done.